src/std/misc/func.ss

```diff
@@ -243,3 +264,32 @@
     (if res
       #t
       (begin (set! res (pred v)) res))))
+
+;; pred-every-of returns a predicate which returns true if all predicates match.
+;; If preds contains a non-predicate, it is transformed into one using equal?
+;; as test if not overridden by the test: keyword.
+;;
+;; Example:
+;;  ((pred-every-of [number? fixnum?]) 2) => #t
+(def (pred-every-of preds test: (test equal?))
+  (if (pair? preds)
+    (begin
+      (set! preds (map (lambda (p) (if (procedure? p) p (cut test p <>))) preds))
+      (lambda (v) (andmap (cut <> v) preds)))
+    pred-true))
+
+;; pred-any-of returns a predicate which returns true if one predicate matches.
+;; If preds contains a non-predicate, it is transformed into one using equal?
+;; as test if not overridden by the test: keyword.
+;;
+;; Example:
+;;  ((pred-any-of ['a number? "b"]) 2) => #t
+(def (pred-any-of preds test: (test equal?))
+  (if (pair? preds)
+    (begin
+      (set! preds (map (lambda (p) (if (procedure? p) p (cut test p <>))) preds))
+      (lambda (v) (ormap (cut <> v) preds)))
```

Comment:
same here, no reason to `set!` (and create an implicit box).

Reply:
Ah, thanks for the info.